Registration plugin

pleandre opened the pull request:
Registration plugin: works well in windows and visual studio. Compile in Ubuntu but still has a problem while starting a registration (symbol lookup error). Tested with nifti reg 1.3.9 and visual studio 2013.

Commit:
First commit: Registration Plugin

Commit:
First Commit: Registration

Commit:
Bug correction and code re-organisation

Commit:
Remove MainWindow requirement for NiftiREg

Commit:
Registration plugin: source code simplification, output in a directory, etc.

shakes76 commented on plugin/registration/CMakeLists.txt:
```diff
@@ -0,0 +1,67 @@
+    set(NIFTI_REG_DIR /usr/local CACHE PATH "NIFTI REG - v1.3.9 - Base Directory")
+ENDIF(UNIX)
+MARK_AS_ADVANCED(NIFTI_REG_DIR)
+message("Using NIFTI REG 1.3.9 from ${NIFTI_REG_DIR}")
```
Remove version name here. Maybe different version.

shakes76 commented on plugin/registration/milxQtRegistrationNiftiReg.h:
```diff
@@ -0,0 +1,89 @@
+	int aladin(ParamsAladin params);
+	int average(QString outputName, QStringList filenames);
+
+	void cpp2def_async(ParamsF3D params);
```
Add documentation for each public member of this class.

shakes76 commented on plugin/registration/milxQtRegistrationImage.cpp:
```diff
@@ -0,0 +1,305 @@
+#include "milxQtRegistrationImage.h"
```
Can you rename this class to just milxQtRegistration? We may want to create a milxQtRegistrationImage class later that overloads the milxQtImage class to add custom registration enable menus etc.

shakes76 commented:
Great plugin Pierre. Only minor comments (see above). Could you also ensure that the sources have the same indents and code style as milxQt sources (4 spaces for tabs etc.). Probably use Astyle to do this. Thanks.

Commit:
Corrections from the first merge review:
Add documentation
Rename class milxQtRegistrationImage to milxQtRegisitration
Remove version name in CMake list
Uniform indentation, etc.. (AStyle)

Commit:
Compilation problem in ubuntu solved

shakes76 commented:
I have created a topic branch for testing purposes for the moment called topic-registration. This will be an ongoing topic branch for the registration plugin. This pull request will be updated until the plugin is released.

Commit:
Added Elastix and ITK algorithm
Bug correction

Commit:
Corrections for Ubuntu

Commit:
Corrections compilation Ubuntu

shakes76 commented:
Thanks for the fixes. Updated topic-registration branch to this version. Compiles fine on OpenSUSE as well.

Commit:
Similarities added

Commit:
Update to Nifti 1.4

Commit:
Correction: missing some #ifdef USE_NIFTI_REG

Commit:
Security added : (control file formats)

shakes76 commented:
I have merged the current code with the latest master and added fixes for more general image types in the topic-registration branch. The work on this pull request is completed. See the topic-registration branch for the latest version of the plugin. The plugin will be included in the next major release of SMILI. Thanks Pierre.
